### ISSUE TITLE
Admin can deinstall patches out of sequence

### ISSUE THREAD

**Issue by sbulen** [SimpleMachines/SMF#7595]
#### Description
Given that 2.1 makes "emulating" an earlier version so easy, you can deinstall SMF point release patches out of order.  

And...  If you do so, _and click thru the warnings_, you really really really break things.  The only fix as near as I can tell is to reload a fresh set of source & reset smfVersion in the DB.  

Bad things happen.  

Low priority enhancement.  But should probably be fixed at some point.  I bet we can find a way to identify the SMF patches and disable the emulation, thus forcing the admin to back them out in reverse order, which works.

![patches](https://user-images.githubusercontent.com/23568484/199583701-bdc14c68-f372-46d5-bbe4-5564727447e7.png)

**Comment by DiegoAndresCortes** [comment]
One possible solution could be to auto delete **patches** after installation, so the admins would _mostly_ always know what patch they are manually reuploading

**Comment by DiegoAndresCortes** [comment]
And there's still the additional issue: If they manually emulate, the forum will tell them to update the forum.
This is worrying because clicking on this link will auto download and attempt to patch the forum which will of course throw errors on _every_ single operation because the forum is in fact, **not outdated**.

**Comment by sbulen** [comment]
Yep.  We would need to find a way to make the patches "emulation proof"...  

I think I would like to keep the prior patches around so you can go back a few if needed, though.  You should be able to, as long as the order is preserved.

**Comment by jdarwood007** [comment]
I would say no changes.  They are making a risk and if the uninstall isn't clean, they will be prompted it will fail to uninstall cleanly.

If this really has to go through, prevent emulation on all packages with the "smf:" prefix.

**Comment by sbulen** [comment]
I hear you.  

What really bugs me about this, though, is it's really different.  In most instances, yeah, you might clobber a file or two if you click thru the warnings deinstalling a mod.  In this instance, you are likely clobbering a LOT of CORE files.  Also, the smfVersion gets set backwards, where the file versions will not (won't match the pattern).  So, you have a guaranteed version lockout, and you almost certainly have LOTS of core files now corrupt.  It's far worse than a normal click thru the warnings.  

And I can definitely see folks saying "I don't need those anymore!"...  

Yeah, we would just need something simple, that keys on the smf: prefix.  Maybe matches a "patch" pattern, to distinguish it from smf mods.

**Comment by jdarwood007** [comment]
we should then maybe switch all the cust team mods/themes over to "smfCustom:".  We can add the ability of to the customize site to protect prefixes and restrict the "smf:" prefix then.
